### PR TITLE
Added ApplicationProtocols support to the TLS options to support AWS MQTT on port 443

### PR DESCRIPTION
**Pull request by henning-krause**
As the title describes, this PR adds support for MQTT connections to AWS using 443 as described on this [page](https://docs.aws.amazon.com/iot/latest/developerguide/protocols.html).

This feature is only available with .NET Core 3.1 onwards, so I've used conditional compilation.

Using this code, a connection can then be made with an AWS Thing:

```cs
var factory = new MqttFactory();
var mqttClient = factory.CreateMqttClient();

var connectOptions = new MqttClientOptionsBuilder()
	.WithClientId("your-client-id")
	.WithTcpServer("your-endpoint.iot.eu-central-1.amazonaws.com", 443)
	.WithProtocolVersion(MqttProtocolVersion.V311)
	.WithTls(new MqttClientOptionsBuilderTlsParameters
	{
		UseTls = true,
		Certificates = new [] {clientCert},
		ApplicationProtocols = new List<SslApplicationProtocol> {new SslApplicationProtocol("x-amzn-mqtt-ca")}
	})
	.Build();

var conResult = await mqttClient.ConnectAsync(connectOptions, CancellationToken.None);
```

### CODE REVIEW

File: Source/MQTTnet/Client/Options/MqttClientOptionsBuilderTlsParameters.cs [hunk 1]
```diff
@@ -27,7 +27,11 @@ public class MqttClientOptionsBuilderTlsParameters
         public IEnumerable<X509Certificate> Certificates { get; set; }
 #endif
 
-        public bool AllowUntrustedCertificates { get; set; }
+        #if NETCOREAPP3_1
```
Comment (chkr1011) [inline Source/MQTTnet/Client/Options/MqttClientOptionsBuilderTlsParameters.cs:30]: Please fix all the formatting issues (intention, missing empty lines, missing spaces in get;set etc.).

File: Source/MQTTnet.AspnetCore/MQTTnet.AspNetCore.csproj [hunk 2]
```diff
@@ -23,10 +23,12 @@
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="System.IO.Pipelines" Version="4.7.2" />
```
Comment (chkr1011) [inline Source/MQTTnet.AspnetCore/MQTTnet.AspNetCore.csproj:26]: This is no longer needed when upgrading the SDK.

**Review by chkr1011 — COMMENTED** [review]
@JanEggers Could you also please have a look,.

**Comment by henning-krause** [comment]
@chkr1011 I think I've resolved all of your issues with the latest commit.

File: Source/MQTTnet/MQTTnet.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;netstandard2.0;netstandard2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard1.3;netstandard2.0;netstandard2.1;netcoreapp3.1</TargetFrameworks>
```
Comment (chkr1011) [inline Source/MQTTnet/MQTTnet.csproj:4]: Please check if the new DLLs are properly added to the nuget package. I don't know if there is a new entry required or not.

**Comment by henning-krause** [comment]
@chkr1011  I've added the nuget instructions in my latest commit. I've verified that the .net core 3.1 dll is in the nuget. 

Can you review the changes?

**Comment by henning-krause** [comment]
@chkr1011 Any news on this PR?

File: Build/build.ps1 [hunk 4]
```diff
@@ -4,7 +4,7 @@ if ([string]::IsNullOrEmpty($assemblyVersion)) {$assemblyVersion = "0.0.1"}
 if ([string]::IsNullOrEmpty($nugetVersion)) {$nugetVersion = "0.0.1"}
 
 $vswhere = ${Env:\ProgramFiles(x86)} + '\Microsoft Visual Studio\Installer\vswhere'
-$msbuild = &$vswhere -products * -requires Microsoft.Component.MSBuild -latest -find MSBuild\**\Bin\MSBuild.exe
+$msbuild = &$vswhere -products * -requires Microsoft.Component.MSBuild -latest -find MSBuild\**\Bin\MSBuild.exe -prerelease
```
Comment (chkr1011) [inline Build/build.ps1:7]: I am not happy with this -prerelease thing. Why is this required? Do we maybe need to wait for a newer release of .NET Core?

**Review by chkr1011 — CHANGES_REQUESTED** [review]

File: Build/build.ps1 (hunk 4 above)
Comment (henning-krause) [reply Build/build.ps1:7]: I needed this because I'm using only preview version of Visual Studio on my machine. We can remove this if it bothers you. It's not required for this PR.

Comment (chkr1011) [reply Build/build.ps1:7]: Yes please remove.

Comment (henning-krause) [reply Build/build.ps1]: Done

**Comment by abduelhamit** [comment]
What prevents this from being compatible with .NET Standard 2.1?

**Comment by JanEggers** [comment]
microsoft provides the option just in netcoreapp :(
